R/fct_helpers.R

```diff
@@ -38,74 +54,320 @@ update_query <- function(active_table, column_name, value, rowid) {
+                                 check_condition = NULL,
+                                 specify_condition_check_condition = NULL,
+                                 collate = NULL,
+                                 collation_type_collate = NULL,
+                                 foreign_key = NULL,
+                                 foreign_table_foreign_key = NULL,
+                                 foreign_column_foreign_key = NULL,
+                                 on_update_foreign_key = NULL,
+                                 on_delete_foreign_key = NULL,
+                                 match_foreign_key = NULL,
+                                 defer_first_foreign_key = NULL,
+                                 defer_second_foreign_key = NULL) {
+  res <- paste0('"', column_name, '" "', data_type, '" ')
+  
+  if (isTRUE(primary_key)) {
+    res <- paste0(res, "PRIMARY KEY ")
+    
+    if (sort_order_primary_key != "")
+      res <- paste0(res, sort_order_primary_key, " ")
+    
+    if (on_conflict_primary_key != "")
+      res <-
+        paste0(res, "ON CONFLICT ", on_conflict_primary_key, " ")
+    
+    if (isTRUE(autoincrement_primary_key))
+      res <- paste0(res, "AUTOINCREMENT ")
+    
+  }
+  
+  if (isTRUE(unique)) {
+    res <- paste0(res, "UNIQUE ")
+    
+    if (on_conflict_unique != "")
+      res <- paste0(res, "ON CONFLICT ", on_conflict_unique, " ")
+  }
+  
+  if (isTRUE(not_null)) {
+    res <- paste0(res, "NOT NULL ")
+    
+    if (on_conflict_not_null != "")
+      res <- paste0(res, "ON CONFLICT ", on_conflict_not_null, " ")
+    
+  }
+  
+  if (isTRUE(default)) {
+    res <- paste0(res, "DEFAULT ")
+    
+    if (!is.na(as.numeric(default_value_default))) {
+      res <- paste0(res, "(", default_value_default, ") ")
+    }
+    else{
+      res <- paste0(res, "'", default_value_default, "' ")
+    }
+    
+  }
+  
+  if (isTRUE(check_condition)) {
+    res <-
+      paste0(res, "CHECK (", specify_condition_check_condition, ") ")
+    
+  }
+  
+  if (isTRUE(collate)) {
+    res <- paste0(res, "COLLATE ", collation_type_collate, " ")
+    
+  }
+  
+  if (isTRUE(foreign_key)) {
+    res <- paste0(
+      res,
+      "REFERENCES \"",
+      foreign_table_foreign_key,
+      "\" (",
+      foreign_column_foreign_key,
+      ") "
+    )
+    
+    if (on_update_foreign_key != "")
+      res <- paste0(res, "ON UPDATE ", on_update_foreign_key, " ")
+    
+    if (on_delete_foreign_key != "")
+      res <- paste0(res, "ON DELETE ", on_delete_foreign_key, " ")
+    
+    if (match_foreign_key != "")
+      res <- paste0(res, "MATCH ", match_foreign_key, " ")
+    
+    if (defer_first_foreign_key != "")
+      res <- paste0(res, defer_first_foreign_key, " ")
+    
+    if (defer_second_foreign_key != "")
+      res <-
+        paste0(res, "INITIALLY ", defer_second_foreign_key, " ")
+    
+  }
+  
+  return(res)
+}
+
+# Create a new table.
+
+create_table_query <- function(table_name = NULL,
```

Comment:
Separate out the functions for creating SQL to a differnet script than fct_helpers